client/post-editor/editor-diff-viewer/index.jsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { get, map } from 'lodash';
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import getPostRevision from 'state/selectors/get-post-revision';
+import getPostRevisionChanges from 'state/selectors/get-post-revision-changes';
+import { normalizeForEditing } from 'state/selectors/utils/revisions';
+
+const EditorDiffViewer = ( { contentChanges, revision } ) => (
+	<div className="editor-diff-viewer">
+		<h1 className="editor-diff-viewer__title">
+			{ get( revision, 'title' ) }
+		</h1>
+		<div className="editor-diff-viewer__content">
+			{ map( contentChanges, ( change, changeIndex ) => {
+				const changeClassNames = classNames( {
+					'editor-diff-viewer__additions': change.added,
+					'editor-diff-viewer__deletions': change.removed,
+				} );
+				return (
+					<span className={ changeClassNames } key={ changeIndex }>
+						{ change.value }
+					</span>
+				);
+			} ) }
+		</div>
+	</div>
+);
+
+EditorDiffViewer.propTypes = {
+	contentChanges: PropTypes.array,
+	postId: PropTypes.number,
+	revision: PropTypes.object,
+	selectedRevisionId: PropTypes.number,
+	siteId: PropTypes.number,
+};
+
+export default connect(
+	( state, ownProps ) => ( {
+		contentChanges: getPostRevisionChanges( state, ownProps.siteId, ownProps.postId, ownProps.selectedRevisionId ),
+		revision: normalizeForEditing(
+			getPostRevision( state, ownProps.siteId, ownProps.postId, ownProps.selectedRevisionId )
+		),
```

Comment:
Just like `getPostRevision`, `normalizeForEditing` should probably be cached via `createSelector`. This would avoid the cost of deeply cloning revision objects, and it would unlock `connect`'s `pure` behavior.